### ISSUE TITLE
RCTMGLMapView does not exist for iOS v6.0.0-alpha2

### ISSUE THREAD

**Issue by kevinsperrine** [nitaliano/react-native-mapbox-gl#658]
I can successfully build and use the example app, but when I attempt to integrate a simple map in my own test app by linking the module the build completes successfully, but fails to run on iOS because the MapView isn't building properly. The android version works great. Can you point me in the right direction here?
```
2017-10-04 23:16:29.765 [warn][tid:com.facebook.react.JavaScript] Warning: Native component for "RCTMGLMapView" does not exist
2017-10-04 23:16:29.767 [error][tid:com.facebook.react.JavaScript] undefined is not an object (evaluating 'MapboxGL.UserTrackingModes')
2017-10-04 23:16:29.769 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: undefined is not an object (evaluating 'MapboxGL.UserTrackingModes')
```

```
// using versions
buymore:aramis chuck$ node --version
v8.4.0
buymore:aramis chuck$ npm --version
5.3.0
buymore:aramis chuck$ yarn --version
1.1.0
buymore:aramis chuck$ react-native --version
react-native-cli: 2.0.1
react-native: 0.49.1
```

**Comment by nitaliano** [comment]
I've seen that error when I don't link libRCTMGL.a under build phases, the v6 install instructions might be missing this step I'll update it today.

**Comment by kevinsperrine** [comment]
Thanks, @nitaliano. When I add libRCTMGL.a to the Link Binary with Libraries build phase I now get errors about a duplicate interface definition for the RCTSource class when compiling CameraUpdateItem.m and MGLModule.m.

![screenshot 2017-10-05 10 08 58](https://user-images.githubusercontent.com/609466/31234837-9166badc-a9b5-11e7-86c3-c87c7f92f918.png)

**Comment by nitaliano** [comment]
hmm would you be able to post a screen of everything you have linked to the project? I'll run through setting it up on a project again and seeing if I get this error.

**Comment by kevinsperrine** [comment]
@nitaliano Sure thing. Is this sufficient? 

![screenshot 2017-10-05 11 22 29](https://user-images.githubusercontent.com/609466/31238418-80e27aa2-a9bf-11e7-83d8-85a041d764c5.png)

```
# Package.json
{
  "name": "aramis",
  "version": "0.1.0",
  "private": true,
  "devDependencies": {
    "babel-core": "6.26.0",
    "babel-jest": "^21.2.0",
    "babel-preset-env": "^1.6.0",
    "babel-preset-react-native": "^4.0.0",
    "babel-preset-react-native-stage-0": "^1.0.1",
    "jest": "^21.2.1",
    "react-test-renderer": "^16.0.0"
  },
  "scripts": {
    "start": "react-native start",
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "test": "jest"
  },
  "jest": {
    "preset": "react-native",
    "transform": {
      "^.+\\.js$": "babel-jest"
    },
    "transformIgnorePatterns": [
      "node_modules/(?!react-native|react-navigation)/"
    ]
  },
  "dependencies": {
    "@mapbox/react-native-mapbox-gl": "^6.0.0-alpha2",
    "prop-types": "^15.6.0",
    "react": "^16.0.0",
    "react-native": "^0.49.1",
    "react-navigation": "^1.0.0-beta.13"
  }
}
```

**Comment by nitaliano** [comment]
@kevinsperrine yes it is 👍

**Comment by kevinsperrine** [comment]
@nitaliano FWIW, I can recreate this with:

1. react-native init mapboxtest
2. cd mapboxtest/
3. yarn add @mapbox/react-native-mapbox-gl@6.0.0-alpha2
4. open ios/mapboxtest.xcodeproj
5. add `node_modules/@mapbox/react-native-mapbox-gl/ios/Mapbox.framework` to Embedded Binaries w/ Copy items if needed 
6. add `node_modules/@mapbox/react-native-mapbox-gl/ios/RCTMGL.xcodeproj` to the Libraries
7. add `$(PROJECT_DIR)/../node_modules/@mapbox/react-native-mapbox-gl/ios` to the framework search paths
8. add a new run script build phase containing `bash "${BUILT_PRODUCTS_DIR}/${FRAMEWORKS_FOLDER_PATH}/Mapbox.framework/strip-frameworks.sh"`
9. add `libRCTMGL.a` to the `Link Binary with Libraries` Build Phase
10. Attempt to build and run the project on simulator.

**Comment by frankrowe** [comment]
I was able to get it to build on iOS by linking `libRCTMGL.a`

**Comment by kevinsperrine** [comment]
@frankrowe Thanks. In digging deeper, it looks like react-native added a RCTSource into core on [version 0.49](https://github.com/facebook/react-native/blob/cd81eff5b891a0b5c528b4315423f157100b67e8/React/Base/RCTJavaScriptLoader.mm#L25). I too was able to get iOS to compile after dropping react-native back to version 0.48.4

cc @nitaliano

**Comment by nitaliano** [comment]
@kevinsperrine ha, great there's also a class called `RCTLayer` I'll go through and namespace them to `RCTMGlLayer` and `RCTMGLSource`. Thanks for digging into this!

I'll publish a new version to npm later today

**Comment by kevinsperrine** [comment]
Thanks, @nitaliano. Great work on this! I'm really excited about this v6 rewrite.

**Comment by nitaliano** [comment]
@kevinsperrine I just pushed the changes to `v6`, updated the example app to use RN 0.49 and published `@mapbox/react-native-mapbox-gl@6.0.0-alpha3` to npm.

I'm going to close this issue out, feel free to create another issue if you run into any problems 👍
